File: frontend/src/constants/fileTypes.ts
/** Keep in sync with backend app/services/file_types.py */

export const TEXT_FILE_EXTENSIONS = ['.pdf', '.docx', '.md', '.txt', '.xlsx', '.xls'] as const;

export const IMAGE_FILE_EXTENSIONS = ['.png', '.jpg', '.jpeg', '.webp', '.gif', '.bmp'] as const;

export const ALLOWED_FILE_EXTENSIONS = [
  ...TEXT_FILE_EXTENSIONS,
  ...IMAGE_FILE_EXTENSIONS,
] as const;

export const FILE_ACCEPT_ATTR = [
  ...ALLOWED_FILE_EXTENSIONS,
  'application/pdf',
  'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
  'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
  'application/vnd.ms-excel',
  'image/png',
  'image/jpeg',
  'image/webp',
  'image/gif',
  'image/bmp',
].join(',');

export function isAllowedUploadFile(file: File): boolean {
  const name = file.name.toLowerCase();
  return ALLOWED_FILE_EXTENSIONS.some((ext) => name.endsWith(ext));
}

export function isImageFileType(fileType: string): boolean {
  return ['png', 'jpg', 'jpeg', 'webp', 'gif', 'bmp'].includes(fileType.toLowerCase());
}

File: backend/app/services/file_types.py
"""Central registry of supported upload file types and preview behavior."""

from __future__ import annotations

# Text extracted into chunks for RAG + text preview
TEXT_EXTRACTABLE_TYPES: frozenset[str] = frozenset({
    "pdf", "docx", "md", "txt", "xlsx", "xls",
})

# Visual preview via original file download (also get minimal text for RAG)
IMAGE_TYPES: frozenset[str] = frozenset({
    "png", "jpg", "jpeg", "webp", "gif", "bmp",
})

DEFAULT_ALLOWED_EXTENSIONS: frozenset[str] = TEXT_EXTRACTABLE_TYPES | IMAGE_TYPES

CONTENT_TYPES: dict[str, str] = {
    "pdf": "application/pdf",
    "docx": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    "md": "text/markdown; charset=utf-8",
    "txt": "text/plain; charset=utf-8",
    "xlsx": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
    "xls": "application/vnd.ms-excel",
    "png": "image/png",
    "jpg": "image/jpeg",
    "jpeg": "image/jpeg",
    "webp": "image/webp",
    "gif": "image/gif",
    "bmp": "image/bmp",
}


def preview_kind(file_type: str) -> str:
    """Return preview mode: image, pdf, or text (chunk-based)."""
    ext = file_type.lower()
    if ext in IMAGE_TYPES:
        return "image"
    if ext == "pdf":
        return "pdf"
    return "text"


def content_type_for(file_type: str) -> str:
    return CONTENT_TYPES.get(file_type.lower(), "application/octet-stream")


def human_labels_zh() -> str:
    return "PDF、Word(DOCX)、Excel(XLS/XLSX)、Markdown、TXT、图片(PNG/JPG/WEBP/GIF/BMP)"
